### PR TITLE
Don't try to normalise the channel if we don't have data for a product.

### PR DESCRIPTION
**Pull request by jgraham**

### CODE REVIEW

**Review by gsnedders — CHANGES_REQUESTED** [review]
Please add tests so we don't regress `wpt run`. We should make sure `setup_wptrunner` can work with all known products.

**Comment by gsnedders** [comment]
@jgraham any idea what caused this to fail CI? the tests that fail are all from @jugglinmike's #10998.

File: tools/wpt/tests/test_run.py [hunk 1]
```diff
@@ -16,3 +22,22 @@ def test_check_environ_fail(platform):
                 run.check_environ("foo")
 
     assert "wpt make-hosts-file" in excinfo.value.message
+
+
+@pytest.mark.parametrize("product", product_list)
+def test_setup_wptrunner(product):
```
Comment (gsnedders) [inline tools/wpt/tests/test_run.py]: This test seems to change some global state, making later tests fail.

`tox -e py27 -- --no-cov -k 'test_run_firefox or test_setup_wptrunner'` is sufficient to observe a failure; Travis has many more.

**Review by gsnedders — CHANGES_REQUESTED** [review]

**Comment by jgraham** [comment]
I made the requested changes some time ago. @gsnedders please re-review.

**Review by gsnedders — APPROVED** [review]

**Review by Hexcles — COMMENTED** [review]
Found two bugs caused by this PR.

**Comment by foolip** [comment]
Given the regression in https://github.com/web-platform-tests/wpt/pull/13567, what sort of test coverage are we missing here?

**Pull request merged**